orchestration/dap_orchestration/solids.py

```diff
@@ -0,0 +1,133 @@
+
+
+@solid(
+    required_resource_keys={"beam_runner", "refresh_directory"},
+    config_schema={
+        "pull_data_dictionaries": Bool,
+        "end_time": String,
+        "target_class": String,
+        "api_token": String,
+        "output_prefix": String,
+    }
+)
+def extract_records(context: AbstractComputeExecutionContext) -> None:
+    """
+    :return: Returns the path to extracted files.
+    """
+    arg_dict = {
+        "pullDataDictionaries": "true" if context.solid_config["pull_data_dictionaries"] else "false",
+        "outputPrefix": f"{context.resources.refresh_directory}/{context.solid_config['output_prefix']}",
+        "endTime": context.solid_config["end_time"],
+        "target_class": context.solid_config["target_class"],
+        "apiToken": context.solid_config["api_token"],
+        "scala_project": "dog-aging-hles-extraction"
+    }
+    context.resources.beam_runner.run(arg_dict)
+
+
+def _build_extract_config(config: dict[str, str], target_class: str, output_prefix: str) -> dict[str, str]:
+    return {
+        "pull_data_dictionaries": config["pull_data_dictionaries"],
+        "output_prefix": output_prefix,
+        "end_time": config["end_time"],
+        "api_token": config["api_token"],
+        "target_class": target_class,
+    }
+
+
+@configured(extract_records)
+def hles_extract_records(config: dict[str, str]) -> dict[str, str]:
+    return _build_extract_config(
+        config,
+        "org.broadinstitute.monster.dap.hles.HLESurveyExtractionPipeline",
+        f"/extract/hles"
+    )
+
+
+@configured(extract_records)
+def cslb_extract_records(config: dict[str, str]) -> dict[str, str]:
+    return _build_extract_config(
+        config,
+        "org.broadinstitute.monster.dap.cslb.CslbExtractionPipeline",
+        "/extract/cslb"
+    )
+
+
+@configured(extract_records)
+def env_extract_records(config: dict[str, str]) -> dict[str, str]:
+    return _build_extract_config(
+        config,
+        "org.broadinstitute.monster.dap.environment.EnvironmentExtractionPipeline",
+        "/extract/env"
+    )
+
+
+@solid(
+    required_resource_keys={"beam_runner", "refresh_directory"},
+    config_schema={
+        "input_prefix": String,
+        "output_prefix": String,
+        "target_class": String,
+    }
+)
+def transform_records(context: AbstractComputeExecutionContext) -> None:
+    """
+    :return: Returns the path to the transformation output json files.
+    """
+    arg_dict = {
+        "inputPrefix": f'{context.resources.refresh_directory}/{context.solid_config["input_prefix"]}',
+        "outputPrefix": f'{context.resources.refresh_directory}/{context.solid_config["output_prefix"]}',
+        "target_class": context.solid_config["target_class"],
+        "scala_project": "dog-aging-hles-transformation"
+    }
+    context.resources.beam_runner.run(arg_dict)
+
+
+def _build_transform_config(target_class: str, input_prefix: str, output_prefix: str) -> dict[str, str]:
```

Comment:
I'm not sure how useful this function is as an abstraction; it simple takes the inputs and puts them in a dictionary, which we might as well just do in each of the configured functions below. Thoughts?

Reply:
@raaidbroad same pushback I had on `_build_extract_config`